### PR TITLE
Add `strict` mode to consider warnings as errors

### PR DESCRIPTION
**Pull request by piranna**
It would be good improve output to make it more uniform with `mocha` errors, but at least it
does the job :-)

### CODE REVIEW

**Comment by Turbo87** [comment]
the last two commits don't seem to be related to the PR description at all. it would probably be better to open a separate PR for them.

**Comment by Turbo87** [comment]
Also I don't see how a "strict" mode is related to #38. 

IMHO if you want the build to fail for warnings you shouldn't use warnings and use errors instead in your ESLint config file.

**Comment by piranna** [comment]
> IMHO if you want the build to fail for warnings you shouldn't use warnings and use errors instead in your ESLint config file.

I don't have decision here, my employer has decided to use `eslint-config-airbnb` without any modification but on the other hand dictates no warnings are allowed, so only way is to enforce them. I don't think this pull-request do any harm...

I'll try to update the pull-request to only have the first commit.

**Comment by piranna** [comment]
Ok, pull-request fixed, let's discuss about the UI issues later.

File: index.js [hunk 1]
```diff
@@ -26,21 +26,19 @@ function test(p, opts) {
...
-        replaceAll(process.cwd() + '/', '', formatter(report.results))
-      );
-    } else if (
-      warn &&
-      report &&
-      report.warningCount > 0
-    ) {
-      console.log(formatter(report.results));
+    if (report) {
+      if (report.errorCount > 0 || opts.strict && report.warningCount > 0) {
```
Comment (Turbo87) [inline index.js]: this should probably be changed to `report.errorCount > 0 || (opts.strict && report.warningCount > 0)` to make the operator order explicit

Comment (piranna) [inline index.js]: AND has precedence over OR, but I've updated the pull-request to set the order explicit since it's easier to read.

Comment (Turbo87) [inline index.js]: thanks

**Comment by Turbo87** [comment]
you should probably also add some documentation about this option to the README.md file

**Comment by piranna** [comment]
> you should probably also add some documentation about this option to the README.md file

Done :-) I've also done some formatting tasks on the README.md file, too.

**Comment by Turbo87** [comment]
LGTM

@IanVS I think this can be merged

**Comment by IanVS** [comment]
I removed the reference to issue #38 so this PR does not close that issue.  I think there's no harm in adding this option, but I would encourage your employer to rethink their mandates, @piranna, as they seem to be contradictory.  

Another way to turn warnings into errors is to create a wrapper for `eslint-config-airbnb` similar to what I did in [eslint-config-standard-warn](https://github.com/IanVS/eslint-config-standard-warn/blob/183dac19eb185f4f5d9868c1e0a678c5e61ca040/index.js).  This way, you would get the correct error level in your editor, and would not be relying on a config setting in a third-party tool (this one).

**Comment by piranna** [comment]
> I removed the reference to issue #38 so this PR does not close that issue. I think there's no harm in adding this option, but I would encourage your employer to rethink their mandates, @piranna, as they seem to be contradictory.

Somewhat, enforce some guidelines without thinking too much into them just because "it's something to agreed" although anybody agree with them. But it's not the first time I'm in this clue, here at Spain we are a bit back in technology and more specially in management... :-(


> Another way to turn warnings into errors is to create a wrapper for eslint-config-airbnb similar to what I did in eslint-config-standard-warn. This way, you would get the correct error level in your editor, and would not be relying on a config setting in a third-party tool (this one).

Good point, I'll pass it to him, thanks! :-)

**Pull request merged**